Dimmer.cpp

```diff
@@ -1,178 +1,152 @@
+	122, 121, 120, 118, 117, 116, 114, 113, 112, 111,
+	110, 109, 108, 107, 106, 105, 104, 103, 102, 101,
+	100,  99,  98,  97,  96,  95,  94,  93,  92,  91,
+	90,  89,  88,  87,  86,  85,  84,  83,  82,  81,
+	80,  79,  78,  77,  76,  75,  74,  73,  72,  71,
+	70,  69,  68,  67,  66,  65,  64,  63,  62,  61,
+	60,  59,  58,  57,  56,  55,  54,  53,  52,  51,
+	50,  48,  47,  46,  45,  44,  42,  41,  40,  38,
+	36,  34,  32,  30,  28,  26,  24,  22,  18,  12, 6
 };
 
+static DimmerControl* Dimmers[MAX_TRIAC]; // Pointers to all registered dimmer objects
+static uint8_t DimmerCount = 0;           // Number of registered dimmer objects
+static bool started = false;
+
 //ZeroCross Interrupt
 void callZeroCross(){
-    dimmerObject.zeroCross();
-}
-
-//Timer interrupt
-void callTriac(){
-    dimmerObject.triac();
+	for (uint8_t i = 0; i < DimmerCount; i++) {
+		Dimmers[i]->zeroCross();
+	}
 }
 
 //Class Constructor
-Dimmer::Dimmer(){
-    zeroCrosAttached=false;
-    triacAttached=false;
-    triacs=0;
-}
-
-void Dimmer::attachZeroCross(int zc_pin, int inter){
-    zeroCrosAttached=true;
-    pinMode(zc_pin, INPUT);
-    attachInterrupt(inter, callZeroCross, RISING);
-}
-
-void Dimmer::attachTriac(byte pin){
-    triacAttached=true;
-    triacPins[triacs]= pin;
-    triacs++;
-}
-
-int Dimmer::init(){
-    ramp_mode=false;
-    countMode=false;
-    return Dimmer::start();
-}
-
-int Dimmer::initRamp(){
-    ramp_mode=true;
-    countMode=false;
-    return Dimmer::start();
-}
-
-int Dimmer::initCount(byte resolution){
-    ramp_mode=false;
-    countMode=true;
-    countResolution = resolution;
-    return Dimmer::start();
-}
-
-void Dimmer::off(){
-    for(byte i=0; i < triacs; i++){
-        lampSwitch[i]=false;
-    }
-}
+DimmerControl::DimmerControl(uint8_t triacPin, uint8_t value, bool state){
+	this->triacPin = triacPin;
+	this->lampValue = value;
+    this->lampState = state;
 
-void Dimmer::off(byte lamp){
-    lampSwitch[--lamp]=false;
+	//Register dimmer object
+	Dimmers[DimmerCount++] = this;
 }
 
-void Dimmer::on(){
-    for(byte i=0; i < triacs; i++){
-        lampSwitch[i]=true;
-    }
-}
-
-void Dimmer::on(byte lamp){
-    lampSwitch[--lamp]=true;
-}
-
-void Dimmer::toggle(byte lamp){
-    lamp--;
-    lampSwitch[lamp]=!lampSwitch[lamp];
-}
-
-void Dimmer::set(byte lamp, byte value){
-    lamp--;
-    if (ramp_mode){
-        lampValue[lamp]=value;
-    }
-    else{
-        lampValue_ramp[lamp]=value;
-    }
-}
-
-int Dimmer::start(){
-    if (!zeroCrosAttached || !triacAttached) {
-        return 0;
-    }
-    
-    if (!countMode) {
-        // Initialize Timer1
-        Timer1.initialize();
-        // Attach timer 1 to triac function
-        //Called every 50 micro-seconds
-        Timer1.attachInterrupt(callTriac, 50);
-    }
+bool DimmerControl::begin(uint8_t mode, uint8_t resolution){
+    if (mode == RAMP_MODE && resolution == 1)
+        resolution=200; //starting resolution for ramp mode
```

Comment:
Try to use spaces of both sides of the "=", as it's the standard we are mostly using now. Ideally we should adhere to some coding standard for these things.